app/models/state_changers.rb

```diff
@@ -162,33 +156,29 @@ def contents_for(target_state)
 
       # determine list of tubes requiring the state change
       # TODO: why does this check specifically for 'failed' when the FILTER_FAILS_ON is a list with several states?
-      racked_tubes_locations_filtered = labware.racked_tubes.reject { |rt| rt.tube.state == 'failed' }.map(&:coordinate)
+      racked_tubes_locations_filtered =
+        v2_labware.racked_tubes.reject { |rt| rt.tube.state == 'failed' }.map(&:coordinate)
 
       # if no tubes are in the target state then no need to send the contents subset (state changer assumes all
       #  will change)
-      return nil if racked_tubes_locations_filtered.length == labware.racked_tubes.count
+      return nil if racked_tubes_locations_filtered.length == v2_labware.racked_tubes.count
 
       # NB. if all tubes are already in the target state then this method will return an empty array
       # TODO: is this correct behaviour?
```

Comment:
Is this comment still true? is it right for an empty array to be returned?
I think I'd noticed that it was calling the SS api to change state on an empty array which seemed pointless.